native/src/main/java/io/ballerina/stdlib/data/FromString.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.data;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import io.ballerina.stdlib.data.utils.DataUtils;
+
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ *  Native implementation of data:fromStringWithType(string).
+ * 
+ * @since 0.1.0
+ */
+public class FromString {
+
+    public static Object fromStringWithType(BString string, BTypedesc typed) {
+        Type expType = typed.getDescribingType();
+
+        try {
+            return fromStringWithType(string, expType);
+        } catch (NumberFormatException e) {
+            return DataUtils.getError("'string' value '" + string + "' cannot be converted to '" + expType + "'");
+        }
+    }
+
+    private static Object fromStringWithType(BString string, Type expType) {
+        String value = string.getValue();
+
+        switch (expType.getTag()) {
+            case TypeTags.INT_TAG:
+                return stringToInt(value);
+            case TypeTags.FLOAT_TAG:
+                return stringToFloat(value);
+            case TypeTags.DECIMAL_TAG:
+                return stringToDecimal(value);
+            case TypeTags.STRING_TAG:
+                return string;
+            case TypeTags.BOOLEAN_TAG:
+                return stringToBoolean(value);
+            case TypeTags.NULL_TAG:
+                return stringToNull(value);
+            case TypeTags.UNION_TAG:
+                return stringToUnion(string, (UnionType) expType);
+            default:
+                throw new IllegalStateException("Invalid type: " + expType);
```

Comment:
Added this to include the `default` case. It won't be reached since we get compile-time error in the ballerina code.

Reply:
Replace it with returning error f3eb6dcbdbb4f80b0ed09b184702c48565e3212d